### PR TITLE
Add uniffi

### PR DESCRIPTION
**Pull request by mfachal**
# Add Uniffi for python bindings

## Description

Adds uniffi for python bindings for stark lib.

## Type of change

Please delete options that are not relevant.

- New feature

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] This change is an Optimization
  - [ ] Benchmarks added/run

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,3 +17,12 @@ benchmarks:
...
 	cargo criterion --bench ${BENCH}
+
+.PHONY: clean
+clean:
+	cargo clean
+
+.PHONY: build.lib.py
+## Compiles the .dylib and generates a python wrapper
+build.lib.py:
+	cargo run --release --bin lambdaworks-stark generate proving-system/stark/src/lambdaworks_stark.udl --language python --out-dir target/python/ --no-format
```
Comment (jpcenteno) [inline Makefile:28]: Remember that we have to copy the dylib into `target/python` for it to work

File: proving-system/stark/src/lib.rs [hunk 2]
```diff
@@ -657,3 +657,18 @@ mod test_utils {
...
+    #[error("Integer overflow on an operation with {a} and {b}")]
+    IntegerOverflow { a: u64, b: u64 },
+}
+
+fn add(a: u64, b: u64) -> Result<u64> {
+    a.checked_add(b)
+        .ok_or(ArithmeticError::IntegerOverflow { a, b })
+}
+
+type Result<T, E = ArithmeticError> = std::result::Result<T, E>;
```
Comment (jpcenteno) [inline proving-system/stark/src/lib.rs:672]: This is great for testing, but we shouldn't merge this.

Let's discuss how to do this.

File: proving-system/stark/Cargo.toml [hunk 3]
```diff
@@ -10,3 +10,18 @@ rand = "0.8.5"
...
+[dev-dependencies]
+uniffi = { version = "0.23.0", features = ["bindgen-tests", "cli"]}
+
+[lib]
+crate-type = ["lib", "cdylib"]
+name = "lambdaworks_stark"
+
+[[bin]]
+name = "lambdaworks-stark"
+path = "lambdaworks_stark.rs"
```
Comment (jpcenteno) [inline proving-system/stark/Cargo.toml:27]: I would rename this to "lambdaworks-stark-uniffi"

**Review by jpcenteno — CHANGES_REQUESTED** [review]

**Comment by jpcenteno** [comment]
Python tests are running on the CI https://github.com/lambdaclass/lambdaworks/actions/runs/4409823169/jobs/7726590088#step:5:639

**Comment by MauroToscano** [comment]
This hasn't been active for a while. Closing it.

**Pull request closed without merging**